### ISSUE TITLE
[Analyzer] Unawaited / returned tasks created with a disposable in using block for that disposable

### ISSUE THREAD

**Issue by stephentoub** [dotnet/runtime#78394]
Consider code like the following:
```C#
public Task MyMethodAsync()
{
    using (SomeIDisposable disposable = new SomeIDisposable())
    {
        ...
        return SomethingAsync(disposable); // or fire-and-forget
    }
}
```
There's a high likelihood that this Task-returning SomethingAsync will be using the SomeIDisposable instance long after it's been disposed.  We should consider an analyzer that flags such situations.

**Comment by bartonjs** [comment]
While it can't detect cases where the IDisposable got wrapped into another parameter, it does seem like it provides value for the ones it detected.

It would also be nice if it could detect the dispose via a manual try-finally or direct imperative code (e.g.

```C#
    SomeIDisposable disposable = new SomeIDisposable(); 
    Task t = SomethingAsync(disposable);
    disposable.Dispose();
    return t;
```
)

And it might be further nice if we could generalize it to rent/return patterns like ArrayPool, but that's perhaps getting a bit fantastical (and each would require a custom flow analyzer, or putting some sort of attribute on IDisposable.Dispose() (and all aliasing Close() methods) and ArrayPool.Return)

Category: Reliability
Severity: Warning

**Comment by steveberdy** [comment]
I'd like to be assigned this for .NET 10

**Comment by stephentoub** [comment]
> I'd like to be assigned this for .NET 10

Great, thanks!

**Comment by steveberdy** [comment]
A possible false positive that hasn't been considered would come from the way awaited tasks are considered.
Technically this would fire, even though the task is awaited before the arguments are disposed:
```csharp
var x = new DisposableThing();
var task = DoSomethingAsync(x);
var task2 = ...;

Task.WaitAll(task, task2);
// or
await Task.WhenAll(task, task2);

x.Dispose();
```

**Comment by akoeplinger** [comment]
@steveberdy yeah I think that's one of the cases I hit in https://github.com/dotnet/dotnet/pull/272/commits/95674bbf161299235a10e5578c3d7cdcce610387
